vault: updating helm docs for v0.33.0 release (#2601)
hashicorp/web-unified-docs: content/vault/v2.x/content/docs/deploy/kubernetes/helm/configuration.mdx

    - `repository` (`string: "hashicorp/vault"`) - The name of the Docker image for the containers running Vault.

>     - `tag` (`string: "2.0.2"`) - The tag of the Docker image for the containers running Vault. **This should be pinned to a specific version when running in production.** Otherwise, other changes to the chart may inadvertently upgrade your admission controller.

Reviewer: ```suggestion
    - `tag` (`string: "2.0.2"`) - The tag of the Docker image for the containers
      running Vault. We recommend pinning the tag to a specific version when
      running in production to avoid accidentally upgrading your admission
      controller due to other chart changes.
```
Style correction: write in active voice, avoid "this" as a pronoun

      - `setNodeId` (`boolean: false`) - Set the Node Raft ID to the name of the pod.

      - `redundancyZones` - Vault Enterprise redundancy zones configuration for autopilot. Requires Kubernetes 1.35+ with PodTopologyLabelsAdmission and nodes labeled with `topology.kubernetes.io/zone`.

        - `enabled` (`boolean: false`) - Enable automatic zone detection for Vault Enterprise autopilot redundancy zones. When enabled:
          - The zone label from the node is exposed as `VAULT_REDUNDANCY_ZONE` environment variable
          - The `VAULT_REDUNDANCY_ZONE` placeholder in `server.ha.raft.config` is automatically substituted with the actual zone value
          - The PDB `maxUnavailable` defaults to 1 (instead of ⌊(n-1)/2⌋) to prevent quorum loss
          - Requires `server.ha.enabled=true`, `server.ha.raft.enabled=true`, and the `autopilot_redundancy_zone = "VAULT_REDUNDANCY_ZONE"` placeholder in the Raft configuration
          - See the [redundancy zones documentation](/vault/docs/platform/k8s/helm/enterprise#redundancy-zones) for more details

> 

Reviewer 2: ```suggestion
        - `enabled` (`boolean: false`) - Enable automatic zone detection for Vault Enterprise autopilot redundancy zones. When enabled:
           The zone label from the node is exposed as `VAULT_REDUNDANCY_ZONE` environment variable
           The `VAULT_REDUNDANCY_ZONE` placeholder in `server.ha.raft.config` is automatically substituted with the actual zone value
           The PDB `maxUnavailable` defaults to 1 (instead of ⌊(n-1)/2⌋) to prevent quorum loss
           Requires `server.ha.enabled=true`, `server.ha.raft.enabled=true`, and the `autopilot_redundancy_zone = "VAULT_REDUNDANCY_ZONE"` placeholder in the Raft configuration
           See the [redundancy zones documentation](/vault/docs/platform/k8s/helm/enterprise#redundancy-zones) for more details

```

Author: @siyer-corp making this change causes the lines to be clubbed like this.

<img width="963" height="196" alt="Image" src="https://github.com/user-attachments/assets/994a5de7-532f-434c-b41a-c640d4857b4d" />

But currently it looks like this.

<img width="731" height="324" alt="Image" src="https://github.com/user-attachments/assets/e6d96e79-479a-482f-b9ae-fb7e465f834c" />

>           - See the [redundancy zones example](/vault/docs/platform/k8s/helm/examples/enterprise-with-raft#redundancy-zones) for configuration details

Reviewer: ```suggestion
      - `redundancyZones` - Autopilot redundancy zone definitions. Requires
        Kubernetes 1.35+ with PodTopologyLabelsAdmission and nodes labeled with
        `topology.kubernetes.io/zone`.

        - `enabled` (`boolean: false`) - Enable automatic zone detection for the
          autopilot redundancy zones. To use automatic zone detection, you must
          set `server.ha.enabled=true`, `server.ha.raft.enabled=true`, and
          `autopilot_redundancy_zone = "VAULT_REDUNDANCY_ZONE"` in the Raft
          configuration.
          
          Enabling automatic zone detection:

          - Exposes the zone label from the node as `VAULT_REDUNDANCY_ZONE`.
          - Vault automatically replaces `VAULT_REDUNDANCY_ZONE` in
            `server.ha.raft.config` with the actual zone value.
          - The PDB `maxUnavailable` defaults to 1 (instead of ⌊(n-1)/2⌋) to
            prevent quorum loss.
```
The primary explanation should not be in an example. If there are additional configuration details, they should be a part of the configuration reference. If we need to provide an overview of how the configuration works, or the consequences of a particular configuration, we should add an overview section instead of trying to cover everything with examples and a field description
